### PR TITLE
FXIOS-1196 ⁃ Fix #7526: Chron tabs telemetry

### LINKED ISSUE #7526: FXIOS-1083 ⁃ [Chron Tabs] Add Probes for Chron Tabs A/B Test
**Vision**

As a Product Manager, I want appropriate probes in place to determine if users are happy with the updates tabs tray design

**Acceptance Criteria**

* The following probes have been added to Glean:
  * Event for opening tab tray
  * Event for opening a tab in tab tray
  * Close tab tray
  * Metrics ping should include the # of tabs a user has
* The probes have been implemented in a way that allow us to measure success metrics:
  * Time from opening tab tray to opening a tab
  * Retention
    * New user
    * Existing user
    * Users with 7 or more tabs
* New/changed events are reflected in our external documentation
* Data stewards have signed off on changes

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1083)

### PR DESCRIPTION
**Pull request by dnarcese**
┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1196)

### CODE REVIEW

**Comment by dnarcese** [comment]
@boek for data review
---
# Request for data collection review form
1. What questions will you answer with this data?
Do users find chronological tab tray easier to use
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
* Determine whether chronological tabs has an effect on user or browser behavior.
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
No other alternatives
4. Can current instrumentation answer these questions?
No
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Click tab tray button</td>
    <td>2</td>
    <td>#7526</td>
  </tr>
<tr>
    <td>Click tab</td>
    <td>2</td>
    <td>#7526</td>
  </tr>
<tr>
    <td>Close tab tray</td>
    <td>2</td>
    <td>#7526</td>
  </tr>
</table>
		
6. How long will this data be collected? Choose one of the following:
* I want to permanently monitor this data. @athomasmoz 
7. What populations will you measure?
All channels and all countries.
8. If this data collection is default on, what is the opt-out mechanism for users?
There is a setting in the app to opt-out
9. Please provide a general description of how you will analyze this data.
Determine time from opening tab tray to opening a tab and retention for new users, existing users, and users with 7 or more tabs.
10. Where do you intend to share the results of your analysis?
Mobile team
11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
No

**Comment by dnarcese** [comment]
Data-review r+

# Data Review Form

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, telemetry is documented in `docs/metrics.md`

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, firefox iOS provides data controls in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

This should be re-reviewed w/ @athomasmoz every 6mo

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

Should review all probes for necessity every 6mo w/ @athomasmoz 

9) Does the data collection use a third-party collection tool? 

No

File: Client/metrics.yaml [hunk 1]
```diff
@@ -353,6 +353,39 @@ tabs:
...
       - firefox-ios@mozilla.com
     expires: "2021-07-21"
+  click_tab:
+    type: event
+    description: |
+      Recorded when a user clicks an open tab
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
```
Comment (nbhasin2) [inline Client/metrics.yaml:363]:
```suggestion
      - https://github.com/mozilla-mobile/firefox-ios/pull/7635/
```

File: Client/metrics.yaml [hunk 2]
```diff
@@ -353,6 +353,39 @@ tabs:
...
+      - firefox-ios@mozilla.com
+    expires: "2021-11-03"
+  open_tab_tray:
+    type: event
+    description: |
+      Recorded when a user opens the tab tray
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
```
Comment (nbhasin2) [inline Client/metrics.yaml:374]:
```suggestion
      - https://github.com/mozilla-mobile/firefox-ios/pull/7635/
```

File: Client/metrics.yaml [hunk 3]
```diff
@@ -353,6 +353,39 @@ tabs:
...
+      - firefox-ios@mozilla.com
+    expires: "2021-11-03"
+  close_tab_tray:
+    type: event
+    description: |
+      Recorded when a user closes the tab tray
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/7526
```
Comment (nbhasin2) [inline Client/metrics.yaml:385]:
```suggestion
      - https://github.com/mozilla-mobile/firefox-ios/pull/7635/
```

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**